sentry-ruby/lib/sentry/threaded_periodic_worker.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Sentry
+  class ThreadedPeriodicWorker
+    include LoggingHelper
+
+    def initialize(logger, internal)
+      @thread = nil
+      @exited = false
+      @interval = internal
+      @logger = logger
+    end
+
+    def ensure_thread
+      return false if @exited
+      return true if @thread&.alive?
+
+      @thread = Thread.new do
+        loop do
+          sleep(@interval)
+          run
```

Comment:
I prefer using the generic `run` instead of `flush` as the API contract of this abstract class. Individual subclasses and simply alias `run` to `flush`.